src/Search/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs

```diff
@@ -0,0 +1,176 @@
+        /// <typeparam name="T">
+        /// The type for which fields will be created, based on its properties.
+        /// </typeparam>
+        /// <param name="contractResolver">
+        /// Contract resolver that the <see cref="SearchIndexClient"/> will use.
+        /// This ensures that the field names are generated in a way that is
+        /// consistent with the way the model will be serialized.
+        /// </param>
+        /// <returns>A collection of fields.</returns>
+        public static IList<Field> BuildForType<T>(IContractResolver contractResolver)
+        {
+            var contract = (JsonObjectContract) contractResolver.ResolveContract(typeof(T));
+            var fields = new List<Field>();
+            foreach (JsonProperty prop in contract.Properties)
+            {
+                DataType dataType = GetDataType(prop.PropertyType, prop.PropertyName);
+
+                var field = new Field(prop.PropertyName, dataType);
+
+                IList<Attribute> attributes = prop.AttributeProvider.GetAttributes(true);
+                foreach (Attribute attribute in attributes)
+                {
+                    IsRetrievableAttribute isRetrievableAttribute;
+                    AnalyzerAttribute analyzerAttribute;
+                    SearchAnalyzerAttribute searchAnalyzerAttribute;
+                    IndexAnalyzerAttribute indexAnalyzerAttribute;
+                    if (attribute is IsSearchableAttribute)
+                    {
+                        field.IsSearchable = true;
+                    }
+                    else if (attribute is IsFilterableAttribute)
+                    {
+                        field.IsFilterable = true;
+                    }
+                    else if (attribute is IsSortableAttribute)
+                    {
+                        field.IsSortable = true;
+                    }
+                    else if (attribute is IsFacetableAttribute)
+                    {
+                        field.IsFacetable = true;
+                    }
+                    else if ((isRetrievableAttribute = attribute as IsRetrievableAttribute) != null)
+                    {
+                        field.IsRetrievable = isRetrievableAttribute.IsRetrievable;
+                    }
+                    else if ((analyzerAttribute = attribute as AnalyzerAttribute) != null)
+                    {
+                        field.Analyzer = AnalyzerName.Create(analyzerAttribute.Name);
+                    }
+                    else if ((searchAnalyzerAttribute = attribute as SearchAnalyzerAttribute) != null)
+                    {
+                        field.SearchAnalyzer = AnalyzerName.Create(searchAnalyzerAttribute.Name);
+                    }
+                    else if ((indexAnalyzerAttribute = attribute as IndexAnalyzerAttribute) != null)
+                    {
+                        field.IndexAnalyzer = AnalyzerName.Create(((IndexAnalyzerAttribute) attribute).Name);
```

Comment:
I missed this earlier. It looks like the cast to `IndexAnalyzerAttribute` is unnecessary if you use the `indexAnalyzerAttribute` variable instead.